feat: migrate recorder service

disururathnayake opened the pull request:
# Description

Migrated recorder-service from an AngularJS CoffeeScript factory to an Angular 17+ @Injectable() service written in TypeScript.


## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- Log in as a Student
- Click "Select Unit" and Select any unit from the dropdown
- Select a task and go to the chat section
- Click on the record icon and record a message
- Send the message and check if your sound is working and the chat is functioning.

**Before**

(Recording the message)
<img width="3016" height="1714" alt="image" src="https://github.com/user-attachments/assets/cbedc00c-a841-4294-8e2b-948f1dec3e21" />

(After sending the recording)
<img width="3022" height="1712" alt="image" src="https://github.com/user-attachments/assets/3403578b-ff68-4142-aa0b-94b75ecdf384" />


**After**

(Recording the message)
<img width="3004" height="1784" alt="image" src="https://github.com/user-attachments/assets/6911a54b-ddaf-41d1-b8c3-44b60f5bcefe" />

(After sending the recording)
<img width="1512" height="982" alt="image" src="https://github.com/user-attachments/assets/92c3bac0-a57f-4466-8a49-2c780c96d5b6" />

## Testing Checklist:

- [x] Tested in latest Chrome
- [x] Tested in latest Safari
- [ ] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

Pasindufdo98 reviewed (approved):
#### LGTM

Hey @disururathnayake,

I tested your component, and both the voice recording and playback functionalities worked as they did in the original CoffeeScript version. I did notice a small thing: after sending a voice recording, the recorded clip remains in the chat bar. I’m not sure if this behavior was present in the previous CoffeeScript/AngularJS implementation.

Overall, the component works well. great job! I’ve attached some screenshots for reference.

Before
<img width="1622" height="952" alt="image" src="https://github.com/user-attachments/assets/61367c12-43cc-4eca-8c76-b1ca269f343b" />

After
<img width="1907" height="1128" alt="image" src="https://github.com/user-attachments/assets/b34ea0fc-45e4-4deb-950d-014e1a66eab7" />

HasinduWelarathne reviewed (approved):
**Works end-to-end**

I pulled the branch and tested the chat voice recorder from the task dashboard. Recording, sending, and playback all behaved as expected and matched the legacy behaviour.

**What I verified**
- Start/stop recording from the chat composer
- Clip uploads, renders in the thread, and plays back with seek
- Multiple clips in the same thread behave independently
- No console errors during record/send/playback
- Composer state looked clean after sending 

**Screenshots**
- While Recording 
<img width="1916" height="1003" alt="Capture 3 while recording" src="https://github.com/user-attachments/assets/5815b3ac-c3c8-4ab1-a53e-cabae3ee0458" />


- After send (two posted audio clips + playback) 
<img width="1916" height="1008" alt="Capture 4 after recording" src="https://github.com/user-attachments/assets/d0252341-04b2-4de4-b894-2d80745c6587" />


Nice work on the migration.

b0ink committed:
chore(release): 10.0.0-44

b0ink committed:
Merge pull request #898 from Shounaks/migration-grade-icon-rebase-onto-origin-8.0.x

Migration grade icon rebase onto origin 8.0.x

b0ink committed:
refactor: check for a valid submission response

returnMarcco committed:
refactor: migrate/unit staff editor 9.x (#933)

* chore: migrate/unit-staff-editor

- delete old Coffeescript file
- delete old template
- remove reference to old Coffeescript file
- link new component
- downgrade new component

* chore: migrate/unit-staff-editor

- port `unit-staff-editor-component` to Typescript from Coffeescript
- port `unit-staff-editor` template to Angular template syntax w/ Angular Material
- adjust parent component `Inputs` and attribute bindings

* Update README.md

* Update README.md

---------

Co-authored-by: Boink <40929320+b0ink@users.noreply.github.com>

b0ink committed:
refactor: unit staff editor material ui (#1001)

* fix: revert staff role on error

* fix: ensure full search for unit role

* chore: fix casing

* refactor: use mat table for staff editor

* chore: remove unused variables

* fix: null check

* refactor: use icon button

* chore: remove unit staff editor coffeescript file

* chore: add back staff editor tooltips

b0ink committed:
fix: check for valid unit

b0ink committed:
feat: add observer only ui

b0ink committed:
chore(release): 10.0.0-45

b0ink committed:
refactor: use students preferred name in greeting chip

b0ink committed:
chore: format

b0ink committed:
refactor: change overdue task wording for assess in portfolio states

b0ink committed:
Merge pull request #1004 from b0ink/fix/preferred-name

refactor: use students preferred name in greeting chip

b0ink committed:
chore: return confirmation modal reference

b0ink committed:
refactor: require confirmation for enabling assess in portfolio for late submissions

b0ink committed:
chore: reword automated assess in portfolio message

b0ink committed:
chore: remove sentence regarding impact of late submission

b0ink committed:
Merge pull request #997 from b0ink/fix/submission-firewall-response

refactor: check for a valid submission response

b0ink committed:
chore(release): 10.0.0-46

b0ink committed:
Merge pull request #1005 from b0ink/fix/assess-in-portfolio-task-due-ui

refactor: change overdue task wording for assess in portfolio states

b0ink committed:
chore: require confirmation for assess in portfolio only option

b0ink committed:
Merge branch '10.0.x' into feat/observer-only

b0ink committed:
Merge pull request #1006 from b0ink/feat/confirm-assess-in-portfolio

feat: confirm assess in portfolio unit setting

b0ink committed:
refactor: migrate unit details editor (#1007)

* refactor: init unit details editor migration

* refactor: migrate unit details editor

* refactor: migrate assess in portfolio confirmation

* refactor: display teaching period dates when selected

* refactor: add draft learning summary hint

* chore: remove old unit details editor files

* chore: remove debug logs

b0ink committed:
Merge branch '10.0.x' into feat/observer-only

b0ink committed:
Merge pull request #1002 from b0ink/feat/observer-only

feat: add observer only ui

b0ink committed:
chore(release): 10.0.0-47

b0ink committed:
refactor: replace range datepickers with three separate datepickers

b0ink committed:
refactor: migrate portfolio grade select step (#1014)

* Frontend migration of portfolio-grade-select-step to Angular 17 component

* Addressed review comments in portfolio-grade-select-step component

* Implement OnInit in PortfolioGradeSelectStepComponent

* replaced bootsrap UI component with Angular material

* chore: bring back deleted lines

* refactor: replace css with tailwind

* chore: revert import order

* chore: revert import order

* chore: revert import order

* chore: revert order

* refactor: simplify grade changing

---------

Co-authored-by: Pasindu Fernando <116358471+Pasindufdo98@users.noreply.github.com>

b0ink committed:
fix: add assess in portfolio tasks as completed tasks in burndown chart

b0ink committed:
feat: add tutor times summary download (#1015)

* feat: add tutor times summary download

* fix: typo

* feat: add calendar view of marking sessions

* refactor: simplify date in filename

* refactor: remove get tutors call

* refactor: custom number of days visible, show today at the end

* refactor: date pickers to filter sessions

* refactor: clean up ui

* refactor: create tutor times analytics component

* refactor: add loading spinner

* refactor: fix timezone issues

* refactor: improve ui

* chore: update endpoint

* feat: toggle hide sessions during tutorials

* chore: remove button

* refactor: implement marking session entity

* chore: remove unused code

* chore: remove comment

b0ink committed:
chore(release): 10.0.0-48

b0ink committed:
feat: display session start and end time

b0ink committed:
chore(release): 10.0.0-49

b0ink committed:
feat: jplag base code (#1010)

* feat: jplag base code

* chore: reword base code description

b0ink committed:
chore(release): 10.0.0-50

b0ink committed:
fix: unlock tasks when prerequisite requires rff and task is in aip state

b0ink committed:
chore(release): 10.0.0-51

b0ink committed:
chore: filter out sessions with less than 1 minute duration

b0ink committed:
fix: filter out students from unit staff editor (#1018)

b0ink committed:
refactor: send timezone of client to the api (#1019)

b0ink committed:
feat: download marking sessions for tutor

b0ink committed:
fix: display tutor name

b0ink committed:
refactor: expand hour segments ui

b0ink committed:
fix: re-enable user filtering on click

b0ink committed:
refactor: collapse related tutorial list

b0ink committed:
feat: campus timezone (#1022)

b0ink committed:
Merge pull request #1020 from b0ink/feat/download-my-marking-sessions

feat: download my marking sessions

b0ink committed:
refactor: enable marking sessions calendar for tutors

b0ink committed:
chore(release): 10.0.0-52

b0ink committed:
refactor: include all submitted tasks in portfolio task list (#1023)

* refactor: include all submitted tasks in portfolio task list

* refactor: fetch tasks included in portfolio from api

* refactor: reword

* refactor: reword

* refactor: show warning if no tasks found

* refactor: wording

* refactor: add error alert

b0ink committed:
chore(release): 10.0.0-53

b0ink committed:
refactor: number list

b0ink committed:
chore: add margin

b0ink committed:
chore(release): 10.0.0-54

b0ink committed:
fix: typo

b0ink committed:
fix: display correct marking session details

b0ink committed:
feat: display task count in inbox (#1025)

* feat: display number of tasks in inbox

* chore: only display task count if not empty

b0ink committed:
refactor: scrollable calendar (#1027)

b0ink committed:
chore(release): 10.0.0-55

b0ink committed:
refactor: require comment for assess in portfolio tasks submitting for feedback (#1026)

* refactor: require comment for assess in portfolio tasks submitting for feedback

* chore: fix comment

* chore: fix indentation

* fix: condition

* chore: update character count text color

* chore: add tooltip

b0ink committed:
fix: ensure grade has been selected

b0ink committed:
chore: display error on failed grade change

b0ink committed:
chore(release): 10.0.0-56

b0ink committed:
refactor: use students preferred name in task comments (#1028)

* refactor: use students preferred name in task comments

* refactor: use users preferred name

b0ink committed:
fix: typo

b0ink committed:
chore: update task status from latest submission details (#1031)

b0ink committed:
fix: enable submission button for 'need help' trigger (#1034)

b0ink committed:
refactor: replace task definition editor with vertical stepper (#1032)

* refactor: replace task def editor with vertical stepper

* chore: capitalise labels

* fix: typo

* fix: typo

b0ink committed:
feat: open similarities from project dashboard (#1035)

* feat: open similarities from project dashboard

* chore: add task similarity view component

b0ink committed:
chore(release): 10.0.0-57

b0ink committed:
refactor: imply portfolio has been submitted (#1037)

b0ink committed:
chore(release): 10.0.0-58

b0ink committed:
chore: ensure student selects grade before confirming learning summary report

b0ink committed:
fix: require comment for new evidence

b0ink committed:
fix: require comment for new evidence

b0ink committed:
chore(release): 10.0.0-59

b0ink committed:
feat: upload grades csv (#1038)

* feat: upload grades csv

* chore: remove debug

b0ink committed:
chore(release): 10.0.0-60

b0ink committed:
refactor: overseer sidekiq (#1029)

* refactor: use sidekiq jobs for pulling overseer images

* refactor: overseer execution script editor

* chore: remove overseer view

* refactor: increase size of script editor

* chore: add icon for incomplete overseer runs

* chore: add typing

* refactor: format assessment comment layout

* refactor: enable test submissions if script exists

b0ink committed:
chore(release): 10.0.0-61

b0ink committed:
feat: restrict assessments to tutors in the same tutorial stream (#1033)

* feat: restrict assessments to tutors in the same tutorial stream

* refactor: unselect tasks that are locked to a tutorial stream

b0ink committed:
chore(release): 10.0.0-62

b0ink committed:
chore: replace doubtfire with ontrack

b0ink committed:
feat: open project dashboard from portfolios view

(#1040)

b0ink committed:
chore(release): 10.0.0-63

b0ink committed:
fix: avoid calling window to open project in new tab

b0ink committed:
refactor: move overseer script url to task definition model

b0ink committed:
refactor: base64 encode overseer script before sending to server

b0ink committed:
Merge pull request #1041 from b0ink/fix/base64-overseer

fix: base64 encode overseer script

b0ink committed:
fix: use new google fonts api for proper weight loading (#1039)

b0ink committed:
chore: specify only time exceeded tasks are updated

b0ink committed:
chore(release): 10.0.0-64

b0ink committed:
fix: typo

b0ink committed:
chore(release): 10.0.0-65

b0ink committed:
fix: correctly set rollover end date

b0ink committed:
feat: discussion prompts (#1042)

* feat: init discussion prompts ui

* refactor: improve tutor discussion ui

* refactor: display discussion prompts in tutor discussion view

* refactor: reword weight to priority

b0ink committed:
refactor: require confirmation for deleting comment (#1059)

b0ink committed:
refactor: render overseer comments as is (#1058)

b0ink committed:
chore(release): 10.0.0-66

b0ink committed:
feat: allow sidekiq web access (#1053)

b0ink committed:
chore(release): 10.0.0-67

b0ink committed:
fix: only trim if valid

b0ink committed:
chore(release): 10.0.0-68

b0ink committed:
feat: display number of stuff notes in tutor discussion

b0ink committed:
feat: attention required task status (#1061)

* feat: init discuss check task status

* refactor: add discuss check status

* refactor: add discuss and check marking button

* refactor: add warning badge to discuss check tasks

* refactor: update discuss check status description

* refactor: update discuss check description

* chore: ensure discuss check status does not unlock requisite tasks

* refactor: rename to off track

* refactor: rename to attention required

* refactor: update status colors

* refactor: rename to attention required internally

* chore: update assess in portfolio color

b0ink committed:
feat: overseer pipeline (#1064)

* feat: init overseer pipeline ui

* chore: init drag and drop steps panel

* feat: init overseer-step crud

* feat: use new monaco editor package

* refactor: clean up ui

* feat: get list of overseer resource files

* chore: allow no input file

* feat: allow partial input output diff check

* feat: allow deleting of overseer steps

* feat: overseer step result model

* refactor: improve overseer report view

* refactor: organise imports

* refactor: retrieve feedback message

* refactor: expose overseer assessment details to task comment

* feat: dropdown for command language

* fix: set fixed height size

* chore: show overseer reports to students

* build: ensure monaco editor is packed correctly

* refactor: lazy load step results for overseer assessments

* refactor: align default feedback messages with the backend

* refactor: improve assessment comment ui

* chore: ignore old assessment script

* refactor: hide test result until report is ready

* fix: use new overseer report component for test submission history

* chore: allow for overseer report refresh

* refactor: reduce icon size

* refactor: auto expand selected overseer report

* chore: enable submission history if project is valid

* chore: delete new step

* chore: set correct overseer steps when changing task definition

* feat: add overseer task status result

* chore: add hint explaining test submissions can be used before enabling overseer

* refactor: hide overseer steps until docker image has been selected

* chore: move overseer editor to original place

* refactor: ensure task definition is saved before adding overseer steps

* chore: show delete and save buttons when step is selected

* refactor: use seconds for overseer timeout

* refactor: show status dropdown only when halting step

* refactor: cleanup

* chore: render diff if overseer step is null

* refactor: remove debug

* chore: show empty overseer reports

b0ink committed:
feat: project task planner gantt chart (#1051)

* feat: init project plan gatt chart

* chore: set compatible version

* chore: update gantt chart package

* refactor: display task items and map prerequisites

* refactor: add task target start and due date

* refactor: show prerequisite for tasks in tooltip

* refactor: save target and due dates

* feat: display task definition dates as baseline items

* chore: remove extension

* fix: create unique baseline item

* refactor: fix timezone issues

* refactor: clean up ui buttons

* refactor: ensure timeline view shows the earliest task

* chore: always show task planner

* refactor: rename project plan

* refactor: minify ui

* chore: confirm changes

* feat: allow target grade changes

* refactor: display tasks target grade

* feat: link to task details

* refactor: remove debug

* chore: remove old task sliders

* refactor: display show date toggle

* chore: add curved links

* fix: save target dates

* chore: reverse prerequisite links

* refactor: reset target dates if task is prerequisite for another task

* chore: fix gantt chart config

* refactor: display warning if prerequisite task ends after dependent task starts

* fix: return false if no linked item

* refactor: highlight prerequisite conflicts

* refactor: move task planner into its own component

* refactor: highlight prerequisites on task hover

* chore: fix prerequisite conflicts

* feat: add status icon to task list

* chore: update bar color

* feat: display task prerequisites modal

* refactor: display dependent tasks

* fix: only use target start and end date if unit allows flexible dates

* refactor: ensure prerequisites are mapped

* chore: only update z-index of lines when hovering over

* feat: task planner card for project dashboard

* chore: add more details

* chore: add tooltip delay

* chore: add description heading

* chore: add back watch command

* refactor: use single endpoint to reset all target dates

* chore: add button to redirect to task planner

* chore: only show dates in tooltip

* refactor: set gantt view dates to latest possible end date

* feat: scroll to and highlight selected task item

* refactor: remove unused code

* refactor: remove unused code

* chore: format

* refactor: assign gantt config variable

* refactor: add oop logic + cleanup duplicated code

* fix: convert correct deadline date string

* chore: import task planner card

b0ink committed:
chore(release): 10.0.0-69

b0ink committed:
chore: allow no expected output file

b0ink committed:
chore: render correct error

b0ink committed:
fix: skip query for unsaved task definition

b0ink committed:
chore: remove required fields

b0ink committed:
chore(release): 10.0.0-70

b0ink committed:
feat: download staff notes csv (#1062)

* feat: download staff notes csv

* chore: restore imports

b0ink committed:
feat: migrate to jplag report viewer v6.3.0 (#1082)

* feat: migrate to jplag report viewer v6.3.0

- now runs on vanilla JPlag report viewer

* chore: update jplag build

* chore: rename variable

b0ink committed:
chore(release): 10.0.0-71

b0ink committed:
fix: null safe user access

b0ink committed:
refactor: render sessions with missing staff

- tutors may have had marking sessions, but then removed from the unit

b0ink committed:
chore(release): 10.0.0-72

b0ink committed:
chore: default jplag src

b0ink committed:
refactor: attempt to scroll jplag comparisons until component is found

b0ink committed:
chore(release): 10.0.0-73

b0ink committed:
chore(release): 10.0.0

b0ink committed:
refactor: update task definition resource files (#1066)

b0ink committed:
chore(release): 10.0.1-0

b0ink committed:
refactor: disable target grade change from project dashboard (#1084)

b0ink committed:
refactor: improve wording and ux for assess in portfolio submissions (#1083)

* refactor: improve wording and ux for assess in portfolio submissions

* chore: increase max width

* refactor: expand on wording

b0ink committed:
feat: tutor notes + mentorship + moderation + escalation (#1068)

* feat: init tutor notes and mentorship

* feat: tutor moderation route

* feat: add moderator actions

* refactor: add confirmation modals

* refactor: move moderation buttons to footer

* feat: filter tutor notes by task

* refactor: mark tutor notes as read

* refactor: view tutor notes via modal

* fix: remove task definition filters for moderation

* chore: add dividers for moderation buttons

* refactor: add tooltip

* refactor: clean up tasks tutor getter

* refactor: rename vars to tutor notes

* refactor: add typing

* refactor: clean up unused code

* fix: check for valid tutor

* feat: init escalation request

* chore: add moderation type field

* refactor: add alternate ui for escalated tasks

* chore: add moderation action type

* feat: feedback appeal modal

* feat: add feedback review requested comment

* chore: revert task-submission-card

* feat: add moderation confirmation modal and allow dismiss all

* chore: dismiss modal

* feat: filter tasks by tutor

* refactor: style action buttons

* chore: rename staff to tutor

* refactor: improve wording

* refactor: improve styling

* chore: add relevant icon

* refactor: update success alert

* chore: increase alert duration

* chore: display tasks tutor

* refactor: sort tutors alphabetically

* chore: remove divider on the right

* refactor: require a comment for review requests

* chore: increase modal max width

b0ink committed:
fix: ensure tutor notes view extends full height

b0ink committed:
chore(release): 10.0.1-1

b0ink committed:
fix: set min width for mentor dropdown

b0ink committed:
refactor: show moderation link to mentors only

b0ink committed:
refactor: group tutor filter by mine and all tutors mentored

b0ink committed:
chore: always show moderation to convenors

b0ink committed:
chore: default to my tutors only if in moderation view

b0ink committed:
Merge pull request #1087 from b0ink/feat/moderation-filters

refactor: group tutor filter by mine and all tutors mentored

b0ink committed:
chore(release): 10.0.1-2

b0ink committed:
chore: disable prerequisite due date validation

b0ink committed:
Merge pull request #1088 from b0ink/chore/disable-prereq-date-validation

chore: disable prerequisite due date validation

b0ink committed:
chore(release): 10.0.1-3

b0ink committed:
feat: allow custom target dates per target grade (#1089)

* feat: allow custom target dates per target grade

* refactor: rename to target dates for clarity

* chore: rename to target date

* refactor: fade datepicker if inherited from default target date

* chore: toast erron on failed task update

* fix: only reset items that are rendered

* fix: ensure baseline items refresh on grade change

* chore: add page heading

* refactor: return per grade due dates if dates are flexible

* chore: toggle due date ui

b0ink committed:
chore: add link to task planner

b0ink committed:
chore: remove debug

b0ink committed:
chore(release): 10.0.1-4

b0ink committed:
feat: per grade start dates (#1090)

* feat: per grade start dates

* feat: display errors when target date is before start date

b0ink committed:
refactor: use projects task dates to get earliest start date

b0ink committed:
chore(release): 10.0.1-5

b0ink committed:
refactor: encode script content to avoid firewall conflicts (#1091)

b0ink committed:
chore(release): 10.0.1-6

b0ink committed:
fix: ensure timestamp is normalised to start of day

b0ink committed:
chore(release): 10.0.1-7

b0ink committed:
feat: overflow marking (#1086)

* feat: init overflow marking

* chore: show overflow link to only staff who can overflow mark

* feat: enable overflow marking permission saving

* feat: feedback overflow unit setting

* chore: track if task is loading submission details

* feat: claim task request

* chore: ensure feedback thresholds are saved

* refactor: disable buttons based on claim conditions

* chore: call claim task request

* fix: display message for task comment errors

* refactor: clean up unit role getter

* refactor: display overflow claim state

* chore: reword label

* chore: add error alert

* fix: check if key exists so it can be reset to null

* refactor: add tooltips, show claim button in box

* chore: remove todo

* refactor: add snackbar

* refactor: move task claim to its own component

* feat: display overdue feedback warnings in staff task list

* refactor: reword task claim tooltip

* chore: extend type

b0ink committed:
chore(release): 10.0.1-8

b0ink committed:
fix: typo

b0ink committed:
chore(release): 10.0.1-9

b0ink committed:
feat: display available storage for docker images (#1092)

* feat: display available storage for docker images

* refactor: hide available storage if disabled

b0ink committed:
chore(release): 10.0.1-10

b0ink committed:
chore: increase allowed failed attempts

- Increase poll delay and failure threshold (5) to improve blue-green Caddy switchover buffer

b0ink committed:
fix: ensure deadline is returned

b0ink committed:
chore: sort task planner by start date

b0ink committed:
chore(release): 10.0.1-11

b0ink committed:
refactor: parse submission date with timestamp

b0ink committed:
chore(release): 10.0.1-12

b0ink committed:
feat: add snooze moderation action (#1096)

b0ink committed:
chore(release): 10.0.1-13

b0ink committed:
refactor: remove streamless tutorials (#1093)

* refactor: remove ui to add tutorial with no stream

* refactor: improve tutorial ui

* refactor: add ui to show no tutorials added

b0ink committed:
chore: add tooltip

b0ink committed:
chore: extend tutor notes max width

b0ink committed:
feat: view tutor notes from unit staff editor

b0ink committed:
chore(release): 10.0.1-14

b0ink committed:
chore: ensure all tasks filter is selected by default if task is null

b0ink committed:
chore: view tutor notes from task dropdown menu

b0ink committed:
chore(release): 10.0.1-15

b0ink committed:
chore: require confirmation for removing staff

b0ink committed:
chore(release): 10.0.1-16

b0ink committed:
fix: empty prompts layout

b0ink committed:
chore: always show mark as read button (#1098)

b0ink committed:
chore(release): 10.0.1-17

b0ink committed:
refactor: migrate unit group set editor (#1100)

* chore: init unit group set editor migration

* refactor: migrate unit group set editor

* chore: remove container class

* fix: set correct unit role

* chore: use mat button

* refactor: format

* chore: improve layout of group set manager

* chore: remove period

* fix: allow unlimited space if group set capacity is null

* chore: remove coffeescript files

b0ink committed:
refactor: migrate csv result + upload modal (#1099)

* chore: init csv result modal migration

* refactor: migrate csv result modal

* refactor: improve ui ux

* refactor: migrate csv-upload-modal

* chore: remove csv result and upload modal coffeescript files

* chore: remove comment

b0ink committed:
fix: dont subtract tutor note count for reading your own notes

b0ink committed:
fix: convert emojis in submission comment to string with colons (#1105)

b0ink committed:
chore(release): 10.0.1-18

b0ink committed:
chore: hide tutor notes icon in footer from tutors

b0ink committed:
feat: view submission history + zip file editor (#1104)

* feat: zip file viewer

* refactor: display upload requirement names instead of zipped file names if available

* feat: show file comparisons when selecting two submissions

* chore: ensure single editor sets correct language

* chore: close menu when selecting for comparison

* refactor: add task and student to modal title

* refactor: clean up ui

* fix: set correct submission number

* refactor: reduce duplicated code

* feat: add tree navigation and edit overseer resources

* refactor: show submission id and most recent

* chore: fix icon size

* refactor: clean up logic

* chore: view submission files as staff only

b0ink committed:
chore(release): 10.0.1-19

b0ink committed:
refactor: declutter footer ui (#1106)

* refactor: declutter footer ui

- toggle between moderation and status buttons in moderation view
- move discussed in class button to extra actions menu
- swap complete and discuss buttons

* chore: remove old discussed in class

* chore: remove old status button switch

b0ink committed:
chore(release): 10.0.1-20

b0ink committed:
refactor: task definition editor refactor (#1063)

* refactor: init task definition editor refactor

* refactor: improve task definition editor

* refactor: improve task definition editor ui

* refactor: add back task definition editor buttons

* chore: highlight selected task definition

* chore: move add task button to right side

* chore: keep task badge centered

* chore: center task badge with arrow icon

* chore: add scroll buffer

b0ink committed:
chore(release): 10.0.1-21

b0ink committed:
feat: require discussion before marking complete (#1103)

* feat: require discussion before marking complete

* chore: wording

b0ink committed:
refactor: migrate project dashboard (#1052)

* refactor: progress unit tasks list ui

* chore: fix task list styling

* feat: load task details from url

* refactor: load task details

* refactor: add task comments viewer

* chore: fix tas list panel resize handle

* chore: add todo

* chore: ensure burndown chart is refreshed

* refactor: ensure progress burndown chart extends full height

* refactor: ensure task list updates on target grade change

* refactor: add submission options menu

* feat: inbox dashboard

* chore: use original inbox icon

b0ink committed:
refactor: migrate students list component (#1107)

* refactor: migrate students list component

* chore: import students list controller

* refactor: remove section div

b0ink committed:
Merge branch '9.x' into 11.0.x

b0ink committed:
chore: fix tasks state

b0ink committed:
refactor: migrate upload submission modal

b0ink committed:
Merge pull request #1108 from b0ink/11.0.x

Merge 9.x migrations into 11.0.x (10.0.x base)

b0ink committed:
refactor: remove unused coffeescript files

b0ink committed:
chore: remove portfolio tasks step component

b0ink committed:
Merge pull request #1109 from b0ink/refactor/remove-unused-angularjs

refactor: remove unused coffeescript files

b0ink committed:
chore: used migrated portfolio welcome step component

b0ink committed:
refactor: migrate portfolio review step component

b0ink committed:
chore: rename variable for clarity

b0ink committed:
Merge pull request #1110 from b0ink/refactor/migrate-portfolio-review-step

refactor: migrate portfolio review step component

b0ink committed:
chore: remove analytics service import

b0ink committed:
refactor: migrate portfolio creation state

b0ink committed:
chore: update README

b0ink committed:
Merge pull request #1111 from b0ink/refactor/portfolio-creation-state

refactor: migrate portfolio creation state

b0ink committed:
fix: set task data for project dashboard state

b0ink committed:
chore: direct to new project dashboard state

b0ink committed:
refactor: add task planner and unit ilos card

b0ink committed:
refactor: clean up unit task list ui

b0ink committed:
chore: scrollable project dashboard

b0ink committed:
fix: remove hardcoded chart view size

b0ink committed:
refactor: migrate project groups state

b0ink committed:
chore: set view type in projects parent state

b0ink committed:
Merge pull request #1112 from b0ink/refactor/migrate-project-groups-state

refactor: migrate project groups state

b0ink committed:
refactor: move tutorials state under new projects parent state

b0ink committed:
refactor: move task planner state to new projects parent state

b0ink committed:
chore: remove unused import

b0ink committed:
refactor: migrate timeout state

b0ink committed:
fix: avoid rendering the staff list twice

- prevents inbox queries from being called twice

b0ink committed:
Merge branch '10.0.x' into 11.0.x

b0ink committed:
Merge pull request #1113 from b0ink/refactor/migrate-timeout

refactor: migrate timeout state

b0ink committed:
Merge branch '11.0.x' of https://github.com/doubtfire-lms/doubtfire-web into 11.0.x

b0ink committed:
chore: remove errors module import

b0ink committed:
refactor: migrate inbox states (#1114)

* refactor: migrate inbox state

* chore: add unit task inbox state

* fix: ensure hard refresh doesnt redirect

* refactor: improve inbox state abstraction

* chore: remove inbox coffeescript states

b0ink committed:
refactor: migrate students list state

b0ink committed:
Merge pull request #1115 from b0ink/refactor/migrate-students-list-state

refactor: migrate students list state

b0ink committed:
chore: update migration progress

b0ink committed:
refactor: improve portfolios ui layout

b0ink committed:
chore: move portfolios state to new unit parent state

b0ink committed:
refactor: migrate unit groups state

b0ink committed:
Merge pull request #1116 from b0ink/refactor/migrate-unit-groups

refactor: migrate unit groups state

b0ink committed:
refactor: migrate analytics state (#1118)

* refactor: migrate analytics state

* chore: wrap component in container

* chore: reset indent

* chore: update README

b0ink committed:
chore: update migration progress

b0ink committed:
chore: remove unused feedback component

b0ink committed:
chore: add staff notes view to inbox

b0ink committed:
chore: null safe access

b0ink committed:
refactor: migrate unit admin editor state

b0ink committed:
chore: remove container class

b0ink committed:
Merge pull request #1119 from b0ink/refactor/migrate-unit-editor

refactor: migrate unit admin editor state

b0ink committed:
feat: staff note and similarity indicators

b0ink committed:
chore: remove content-editable angularjs component

b0ink committed:
chore: remove unused outcome-service angularjs component

b0ink committed:
chore: deprecate filters.coffee

b0ink committed:
chore: remove analytics service

b0ink committed:
Merge pull request #1120 from b0ink/refactor/remove-analytics-service

chore: remove analytics service

b0ink committed:
chore: remove google analytics config

b0ink committed:
Merge branch '11.0.x' into migrate/recorder-service-component
